components/Pagination/Pagination.js

```diff
@@ -0,0 +1,132 @@
+  currentPageTooSmall: '"currentPage" cannot be less than 1.',
+  currentPageTooBig: '"currentPage" cannot be larger than "totalPages".',
+};
+
+const getPagination = (currentPage, totalPages) => {
+  // maximum length of the Pagination Bar, should be an odd integer, default is 11
+  const MAX_VISIBLE_ELEMENTS = 11;
+  const ELEMENTS_ON_ONE_SIDE = (MAX_VISIBLE_ELEMENTS - 1) / 2; // 5
+
+  const isTruncatingRequired = totalPages >= MAX_VISIBLE_ELEMENTS;
+
+  const isLeftSideLengthy = currentPage - 1 > ELEMENTS_ON_ONE_SIDE;
+  const isRightSideLengthy = totalPages - currentPage > ELEMENTS_ON_ONE_SIDE;
+
+  const shouldTruncateStart = isTruncatingRequired && isLeftSideLengthy;
+  const shouldTruncateEnd = isTruncatingRequired && isRightSideLengthy;
+
+  const truncateStartOnly = shouldTruncateStart && !shouldTruncateEnd;
+  const truncateEndOnly = !shouldTruncateStart && shouldTruncateEnd;
+
+  let paginationStart;
+  let paginationEnd;
+
+  if (truncateStartOnly) {
+    paginationStart = totalPages - MAX_VISIBLE_ELEMENTS + 3;
+    paginationEnd = totalPages;
+  } else if (truncateEndOnly) {
+    paginationStart = 1;
+    paginationEnd = MAX_VISIBLE_ELEMENTS - 2;
+  } else {
+    paginationStart = shouldTruncateStart ? currentPage - (ELEMENTS_ON_ONE_SIDE - 2) : 1;
+    paginationEnd = shouldTruncateEnd ? currentPage + (ELEMENTS_ON_ONE_SIDE - 2) : totalPages;
+  }
+
+  const paginationLength = paginationEnd - paginationStart + 1;
+
+  return {
+    paginationStart,
+    paginationLength,
+    shouldTruncateStart,
+    shouldTruncateEnd,
+  };
+};
+
+// eslint-disable-next-line react/prop-types
+const PaginationItems = ({ currentPage, totalPages }) => {
+  const {
+    paginationStart,
+    paginationLength,
+    shouldTruncateStart,
+    shouldTruncateEnd,
+  } = getPagination(currentPage, totalPages);
+
+  const PaginationItemArray = [...Array(paginationLength)].map((_, index) => {
+    const page = index + paginationStart;
+    const isCurrent = page === currentPage;
+
+    return <PaginationItem key={page} value={`${page}`} isCurrent={isCurrent} testId={`${page}`} />;
+  });
+
+  return (
+    <>
+      {shouldTruncateStart && (
+        <>
+          <PaginationItem key="1" value="1" testId="1" />
+          <PaginationItem
+            key="seperatorStart"
+            value="&hellip;"
+            isClickable={false}
+            testId="seperatorStart"
+          />
+        </>
+      )}
+      {PaginationItemArray}
+      {shouldTruncateEnd && (
+        <>
+          <PaginationItem
+            key="seperatorEnd"
+            value="&hellip;"
+            isClickable={false}
+            testId="seperatorEnd"
+          />
+          <PaginationItem key={totalPages} value={`${totalPages}`} testId={`${totalPages}`} />
+        </>
+      )}
+    </>
+  );
+};
+
+function Pagination({ currentPage, totalPages }) {
+  const isCurrentPageTooSmall = currentPage < 1;
+  if (isCurrentPageTooSmall && process.env.NODE_ENV !== 'production') {
```

Comment:
Instead of checking `process.env.NODE_ENV !== 'production'` for both, you can just wrap all of the dev error stuff into one conditional:

```javascript
if (process.env.NODE_ENV !== 'production') {
  const isCurrentPageTooSmall = currentPage < 1;
  if (isCurrentPageTooSmall) {
  // etc...

}
```

Reply:
Done :+1: